@xen-orchestra/web/src/components/pool/PoolNetworksTable.vue

```diff
@@ -0,0 +1,321 @@
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+
+import {
+  faAlignLeft,
+  faCaretDown,
+  faCopy,
+  faEdit,
+  faEllipsis,
+  faHashtag,
+  faPowerOff,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { computed, ref, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const props = defineProps<{
+  networks: XoNetwork[]
+  isReady: boolean
+  hasError: boolean
+  selectedRowId: string | null
+}>()
+
+const emit = defineEmits<{
+  rowSelectNetwork: [value: any]
+}>()
+
+const pagination = ref<PaginationPayload>({
+  currentPage: 1,
+  pageSize: 10,
+  startIndex: 0,
+  endIndex: 0,
+})
+
+const { t } = useI18n()
+
+const { records: pifs } = usePifStore().subscribe()
+
+const findPageById = () => {
+  const index = props.networks.findIndex(network => network.id === props.selectedRowId)
+  if (index === -1) return null
+  pagination.value.currentPage = Math.floor(index / pagination.value.pageSize) + 1
+}
+
+watch(
+  () => props.networks,
+  () => {
+    findPageById()
+  }
+)
+
+const selectRow = (item: any, table: string) => {
+  emit('rowSelectNetwork', { item, table })
+}
+
+const searchQuery = ref('')
+
+const filteredNetworks = computed(() => {
+  let filtered = props.networks
+  if (!searchQuery.value) return props.networks.slice(pagination.value.startIndex - 1, pagination.value.endIndex)
+  filtered = props.networks.filter(network =>
+    Object.values(network).some(value => String(value).toLowerCase().includes(searchQuery.value.toLowerCase()))
+  )
+  return filtered.slice(pagination.value.startIndex - 1, pagination.value.endIndex)
+})
+
+const getNetworkVlan = (network: XoNetwork) => {
+  const networkPIFs = pifs.value.filter(pif => network.PIFs.includes(pif.id))
+  if (networkPIFs.length > 0) {
+    return networkPIFs[0].vlan !== -1 ? networkPIFs[0].vlan.toString() : t('none')
+  }
+}
+
+const getLockingMode = (lockingMode: boolean) => {
+  return lockingMode ? t('disabled') : t('unlocked')
+}
+
+const usableIds = computed(() => props.networks.map(network => network.id))
+const { selected, areAllSelected } = useMultiSelect(usableIds)
+const toggleSelect = () => {
+  selected.value = selected.value.length === 0 ? usableIds.value : []
+}
+
+const { visibleColumns, rows } = useTable('networks', filteredNetworks, {
+  rowId: record => record.id,
+  columns: define => [
+    define('checkbox', () => '', { label: '', isHideable: false }),
+    define('name_label', (record: XoNetwork) => record.name_label, { label: computed(() => t('name')) }),
+    define('name_description', (record: { name_description: string }) => record.name_description, {
+      label: computed(() => t('description')),
+    }),
+    define('status', () => '', { label: computed(() => t('pifs-status')) }),
+    define('vlan', () => 'l', { label: 'VLAN' }),
```

Comment:
I don't think the `l` is needed in the getter.
By the way, all terms should be translated:
```suggestion
    define('vlan', () => '', { label: t('vlan') }),
```